cirq-core/cirq/vis/circuit_to_latex_quantikz.py

```diff
@@ -0,0 +1,710 @@
+            - `"index"`: Labels as $0, 1, \dots$
+            - `"qid"`: Labels as the string representation of the `cirq.Qid`
+            - Any other value defaults to `"q"`.
+        show_parameters: A boolean indicating whether gate parameters (e.g.,
+            exponents for `XPowGate`, angles for `Rx`) should be displayed
+            in the gate labels.
+        gate_name_map: An optional dictionary mapping Cirq gate names (strings)
+            to custom LaTeX strings for rendering. This allows renaming gates
+            in the output.
+        float_precision_exps: An integer specifying the number of decimal
+            places for formatting floating-point exponents.
+        float_precision_angles: An integer specifying the number of decimal
+            places for formatting floating-point angles. (Note: Not fully
+            implemented in current version for all angle types).
+
+    Raises:
+        ValueError: If the input `circuit` is empty or contains no qubits.
+    """
+
+    GATE_NAME_MAP = {
+        "Rx": r"R_{X}",
+        "Ry": r"R_{Y}",
+        "Rz": r"R_{Z}",
+        "FSim": r"\mathrm{fSim}",
+        "PhasedXZ": r"\Phi",
+        "CZ": r"\mathrm{CZ}",
+        "CX": r"\mathrm{CX}",
+        "iSwap": r"i\mathrm{SWAP}",
+    }
+
+    def __init__(
+        self,
+        circuit: "cirq.Circuit",
+        *,
+        gate_styles: Optional[Dict[str, str]] = None,
+        quantikz_options: Optional[str] = None,
+        fold_at: Optional[int] = None,
+        custom_preamble: str = "",
+        custom_postamble: str = "",
+        wire_labels: str = "q",
+        show_parameters: bool = True,
+        gate_name_map: Optional[Dict[str, str]] = None,
+        float_precision_exps: int = 2,
+        float_precision_angles: int = 2,
+    ):
+        if not circuit:
+            raise ValueError("Input circuit cannot be empty.")
+        self.circuit = circuit
+        self.gate_styles = gate_styles if gate_styles is not None else GATE_STYLES_COLORFUL1.copy()
+        self.quantikz_options = quantikz_options or ""
+        self.fold_at = fold_at
+        self.custom_preamble = custom_preamble
+        self.custom_postamble = custom_postamble
+        self.wire_labels = wire_labels
+        self.show_parameters = show_parameters
+        self.current_gate_name_map = self.GATE_NAME_MAP.copy()
+        if gate_name_map:
+            self.current_gate_name_map.update(gate_name_map)
+        self.sorted_qubits = self._get_sorted_qubits()
+        if not self.sorted_qubits:
+            raise ValueError("Circuit contains no qubits.")
+        self.qubit_to_index = self._map_qubits_to_indices()
+        self.num_qubits = len(self.sorted_qubits)
+        self.float_precision_exps = float_precision_exps
+        self.float_precision_angles = float_precision_angles
+
+        # Initialize gate maps here to avoid circular import issues
+        self._SIMPLE_GATE_MAP: Dict[Type["cirq.Gate"], str] = {cirq.MeasurementGate: "Measure"}
```

Comment:
You can probably declare these globally if you change all the 'cirq.Gate' to 'ops.Gate'